lib/metadata/src/metadata.cpp

```diff
@@ -0,0 +1,172 @@
+        }
+    } // set
+
+
+    void add(rxComm_t& _comm, const avu& _md, entity_type _et, const std::string& _tgt)
+    {
+        if(const auto ec = mod_avu_meta(_comm, _md, {}, _et, _tgt, "add"); ec < 0) {
+            THROW(ec, fmt::format("metadata add failed for [{}]", _tgt));
+        }
+    } // add
+
+    void remove(rxComm_t& _comm, const avu& _md, entity_type _et, const std::string& _tgt)
+    {
+        if(const auto ec = mod_avu_meta(_comm, _md, {}, _et, _tgt, "rm"); ec < 0) {
+            THROW(ec, fmt::format("metadata remove failed for [{}]", _tgt));
+        }
+    } // remove
+
+    void modify(rxComm_t& _comm, const avu& _md, const avu& _dmd, entity_type _et, const std::string& _tgt)
+    {
+        avu tmp{
+              std::string{"n:"} + _dmd.attribute
+            , std::string{"v:"} + _dmd.value
+            , std::string{"u:"} + _dmd.units
+        };
+
+        if(const auto ec = mod_avu_meta(_comm, _md, tmp, _et, _tgt, "mod"); ec < 0) {
+            THROW(ec, fmt::format("metadata modify failed for [{}]", _tgt));
+        }
+    } // modify
+
+    auto get(rxComm_t& _comm, entity_type _et, const std::string& _tgt) -> std::vector<avu>
+    {
+        namespace fs = irods::experimental::filesystem;
+
+        auto temp = std::string{"SELECT META_{}_ATTR_NAME, META_{}_ATTR_VALUE, META_{}_ATTR_UNITS WHERE {}"};
+
+        const std::map<entity_type, std::string> type_to_string {
+            {entity_type::collection,  "COLL"},
+            {entity_type::data_object, "DATA"},
+            {entity_type::user,        "USER"},
+            {entity_type::resource,    "RESC"}
+        };
+
+        const std::map<entity_type, std::string> type_to_query {
+            {entity_type::collection,  "COLL_NAME = '{}'"},
+            {entity_type::data_object, "COLL_NAME = '{}' AND DATA_NAME = '{}'"},
+            {entity_type::user,        "USER_NAME = '{}'"},
+            {entity_type::resource,    "RESC_NAME = '{}'"}
+        };
+
+        auto where = type_to_query.at(_et);
+
+        std::string where_fmt;
+        if(entity_type::data_object == _et) {
+            fs::path p{_tgt};
+            auto obj  = p.object_name().string();
+            auto coll = p.parent_path().string();
+
+            where_fmt = fmt::format(fmt::runtime(where), coll, obj);
+        }
+        else {
+            where_fmt = fmt::format(fmt::runtime(where), _tgt);
+        }
+
+        auto type_str  = type_to_string.at(_et);
+        auto query_str = fmt::format(fmt::runtime(temp), type_str, type_str, type_str, where_fmt);
+
+        auto res = std::vector<avu>{};
+
+        try {
+            for(auto&& r : query(&_comm, query_str)) {
+                res.push_back(avu{r[0], r[1], r[2]});
+            }
+        }
+        catch(const irods::exception& e) {
+            if(e.code() != CAT_NO_ROWS_FOUND) {
+                THROW(e.code(), e.what());
+            }
+        }
```

Comment:
Add a catch block that handles `std::exception` and `...`.

The `std::exception` catch block can throw `SYS_LIBRARY_ERROR` or `SYS_INTERNAL_ERR`.
The `...` catch block should throw `SYS_UNKNOWN_ERROR`.